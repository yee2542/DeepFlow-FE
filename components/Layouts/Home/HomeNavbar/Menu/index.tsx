import {
  MenuItemStyled,
  NavbarContainerRow,
} from 'Components/Layouts/Home/HomeNavbar/styled';
import { APP_NAVBAR_CONSTANT } from 'Models/menu/constant';
import Link from 'next/link';
import { useRouter } from 'next/router';
import React from 'react';
import pathJoin from 'Utils/path-join';

const Menu: React.FC = () => {
  const router = useRouter();
  return (
    <NavbarContainerRow>
      {APP_NAVBAR_CONSTANT.map(({ key, label, link }) => {
        const isSelecting = '/' + link === router.pathname;
        return (
          <li key={key}>
            <Link href={pathJoin(link)}>
              <MenuItemStyled isSelecting={isSelecting}>{label}</MenuItemStyled>
            </Link>
          </li>
        );
      })}
    </NavbarContainerRow>
  );
};

export default Menu;
