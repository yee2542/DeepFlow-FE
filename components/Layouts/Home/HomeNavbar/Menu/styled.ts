import styled from 'styled-components';

export const MenuPStyled = styled.p<{ isSelecting: boolean }>`
  color: ${(props) => (props.isSelecting ? '#0BB5C2' : 'black')};
  cursor: pointer;
  &:hover {
    color: #0bb5c2;
  }
`;
