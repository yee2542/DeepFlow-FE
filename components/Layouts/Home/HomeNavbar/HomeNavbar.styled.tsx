import styled from 'styled-components';

export const Nav = styled.nav`
  width: 100%;
  height: 55px;
  border-bottom: 2px solid #f1f1f1;
  padding: 0 15px 20px 20px;
  display: flex;
  justify-content: space-between;

  background-color: white;
`;

export const Logo = styled.div`
  padding-top: 15px;
  font-weight: 2em;
`;

export const BtnSideMenu = styled.button`
  background: none;
  border: 0px;
  cursor: pointer;

  @media (min-width: 992px) {
    display: none;
  }
`;

export const Ul = styled.ul`
  list-style: none;
  display: flex;
  flex-flow: row nowrap;

  li {
    padding: 15px 10px;
  }

  @media (max-width: 992px) {
    display: none;
  }
`;

export const Li = styled.li`
  list-style: none;
  margin: 1em 1em 1em 1.4em;
`;
