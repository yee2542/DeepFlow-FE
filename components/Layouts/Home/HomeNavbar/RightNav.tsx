import APP_NAVBAR_CONSTANT from 'Models/menu/app.navbar.constant';
import React from 'react';
import styled from 'styled-components';
import pathJoin from 'Utils/path-join';

const RightNav: React.FC = () => {
  const Ul = styled.ul`
    list-style: none;
    display: flex;
    flex-flow: row nowrap;

    li {
      padding: 18px 10px;
    }

    @media (max-width: 992px) {
      display: none;
    }
  `;
  return (
    <Ul>
      {APP_NAVBAR_CONSTANT.map(({ key, label, link }) => {
        return (
          <li key={key}>
            <a href={pathJoin(link)}>{label}</a>
          </li>
        );
      })}
    </Ul>
  );
};

export default RightNav;
