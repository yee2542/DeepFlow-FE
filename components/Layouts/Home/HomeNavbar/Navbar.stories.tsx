// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Meta, Story } from '@storybook/react/types-6-0';
import HomeNavbar from 'Components/Layouts/Home/HomeNavbar/HomeNavbar';
import React from 'react';
import 'antd/dist/antd.css';

export default {
  title: 'FluentSearch/Navbar',
  component: HomeNavbar,
} as Meta;

const NavbarStory: Story = (args): JSX.Element => <HomeNavbar {...args} />;

// LoggedIn && LoggedOut

export const LoggedOut = NavbarStory.bind({});
