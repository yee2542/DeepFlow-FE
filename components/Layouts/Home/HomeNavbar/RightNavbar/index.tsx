import { UlStyle } from 'Components/Layouts/Home/HomeNavbar/styled';
import { APP_NAVBAR_CONSTANT } from 'Models/menu/constant';
import Link from 'next/link';
import { useRouter } from 'next/router';
import React from 'react';
import pathJoin from 'Utils/path-join';

const RightNav: React.FC = () => {
  const router = useRouter();
  return (
    <UlStyle>
      {APP_NAVBAR_CONSTANT.map(({ key, label, link }) => {
        return (
          <li key={key}>
            <Link href={pathJoin(link)}>
              <p style={{ color: '/' + link === router.pathname ? '#0BB5C2' : 'black' }}>
                {label}
              </p>
            </Link>
          </li>
        );
      })}
    </UlStyle>
  );
};

export default RightNav;
