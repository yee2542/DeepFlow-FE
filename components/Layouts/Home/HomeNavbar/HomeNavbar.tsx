/* eslint-disable @typescript-eslint/explicit-function-return-type */
import { MenuOutlined } from '@ant-design/icons';
import { BtnSideMenu, Nav } from 'Components/Layouts/Home/HomeNavbar/HomeNavbar.styled';
import RightNav from 'Components/Layouts/Home/HomeNavbar/RightNav';
import SideMenu from 'Components/Layouts/Home/HomeNavbar/SideMenu';
import React, { useState } from 'react';
import style from '../styles.module.css';

const HomeNavbar: React.FC = () => {
  const [visible, setVisible] = useState(false);

  const showDrawer = () => {
    setVisible(true);
  };

  const onClose = () => {
    setVisible(false);
  };

  return (
    <div>
      <Nav className={style.navbar}>
        <div className="logo">Deepflow</div>
        <RightNav />

        <BtnSideMenu onClick={showDrawer}>
          <MenuOutlined />
        </BtnSideMenu>
      </Nav>
      <SideMenu onClose={onClose} visible={visible} />
    </div>
  );
};

export default HomeNavbar;
