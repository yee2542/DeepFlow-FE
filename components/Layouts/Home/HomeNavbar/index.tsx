/* eslint-disable @typescript-eslint/explicit-function-return-type */
import { MenuOutlined } from '@ant-design/icons';
import {
  BtnSideMenu,
  Logo,
  MenuDiv,
  Nav,
} from 'Components/Layouts/Home/HomeNavbar/styled';
import RightNav from 'Components/Layouts/Home/HomeNavbar/RightNavbar';
import SideMenu from 'Components/Layouts/Home/HomeNavbar/SideMenu';
import React, { useState } from 'react';

interface HomeNavbarProps {
  visible?: boolean;
}

const HomeNavbar: React.FC<HomeNavbarProps> = (props) => {
  const [visible, setVisible] = useState(props.visible || false);

  const showDrawer = () => {
    setVisible(true);
  };

  const onClose = () => {
    setVisible(false);
  };

  return (
    <div>
      <Nav>
        <Logo>FluentSearch</Logo>
        <RightNav />

        <BtnSideMenu onClick={showDrawer}>
          <MenuDiv>
            <MenuOutlined />
          </MenuDiv>
        </BtnSideMenu>
      </Nav>
      <SideMenu onClose={onClose} visible={visible} />
    </div>
  );
};

export default HomeNavbar;
