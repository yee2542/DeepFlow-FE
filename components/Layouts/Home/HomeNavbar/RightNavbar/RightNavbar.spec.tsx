import { mount } from 'enzyme';
import RightNavbar from 'Components/Layouts/Home/HomeNavbar/RightNavbar';
import React from 'react';
import { Logo } from 'Components/Layouts/Home/HomeNavbar/styled';

describe('Components/Navbar test', () => {
  it('Render correctly', () => {
    const wrap = <RightNavbar />;
    expect(wrap).toMatchSnapshot();
  });

  it('RightNavbar should be exisiting', () => {
    const wrap = mount(<RightNavbar />);
    console.log(wrap.debug());
  });
});
