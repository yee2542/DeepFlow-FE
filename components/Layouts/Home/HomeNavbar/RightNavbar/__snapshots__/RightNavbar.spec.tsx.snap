// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Navbar test Render correctly 1`] = `<RightNav />`;
